### PR TITLE
Added diskcontrollertype to restore points

### PR DESCRIPTION
**Pull request by raktdas**
# ARM (Control Plane) API Specification Update Pull Request 

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I have reviewed the general guidance on the spec PR review process: https://aka.ms/specprreview.
- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data-plane related specifications.
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy).
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can request review from an ARM API Review board.

### ARM API changes review

- If you want for the ARM team to review this PR, you must add the `ARMReview` label. 
- The automation may automatically add the `ARMReview` label, if appropriate.  
  If this happens, proceed according to guidance given in GitHub comments also added by the automation.

### Breaking change review

If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), 
follow the process outlined in the [High-level Breaking Change Process doc](https://aka.ms/breakingchangesprocess#high-level-breaking-change-process).
      
## Getting help

- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For additional help, see https://aka.ms/azsdk/support/spectools.

### CODE REVIEW

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/restorePoint.json [hunk 1]
```diff
@@ -713,24 +689,32 @@
           "items": {
             "$ref": "#/definitions/RestorePointSourceVMDataDisk"
           },
-          "x-ms-identifiers": [
-            "lun"
-          ],
+          "x-ms-identifiers": ["lun"],
           "description": "Gets the data disks of the VM captured at the time of the restore point creation."
+        },
+        "diskControllerType": {
```
Comment (smotwani) [inline specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/restorePoint.json:695]: Pls. indicate that this property is read-only as the user cannot specify this property in the source metadata of restore point resource.

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/restorePoint.json [hunk 2]
```diff
@@ -713,24 +689,32 @@
...
       "description": "Describes the storage profile."
     },
+    "RestorePointSourceVMDiskControllerType": {
+      "type": "string",
+      "enum": ["SCSI", "NVMe"],
+      "x-ms-enum": {
+        "name": "DiskControllerTypes",
+        "modelAsString": true
+      },
+      "description": "Specifies the disk controller type configured for the VM and VirtualMachineScaleSet. This property is only supported for virtual machines whose operating system disk and VM sku supports Generation 2 (https://docs.microsoft.com/en-us/azure/virtual-machines/generation-2), please check the HyperVGenerations capability returned as part of VM sku capabilities in the response of Microsoft.Compute SKUs api for the region contains V2 (https://docs.microsoft.com/rest/api/compute/resourceskus/list). For more information about Disk Controller Types supported please refer to https://aka.ms/azure-diskcontrollertypes."
```
Comment (smotwani) [inline specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/restorePoint.json:709]: Provides the disk controller type configured for the VM. This property is only supported for virtual machines whose operating system disk and VM sku supports Generation 2 (https://docs.microsoft.com/en-us/azure/virtual-machines/generation-2). For more information about Disk Controller Types supported please refer to https://aka.ms/azure-diskcontrollertypes.

**Comment by grizzlytheodore** [comment]
@ArcturusZhang  can you help troubleshoot why Swagger APIView  is failing?

**Review by smotwani — APPROVED** [review]

**Review by kirankumarp-azure — APPROVED** [review]

**Review by grizzlytheodore — APPROVED** [review]

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/restorePoint.json [hunk 3]
```diff
@@ -717,10 +717,27 @@
...
+        },
+        "diskControllerType": {
+          "$ref": "#/definitions/RestorePointSourceVMDiskControllerType",
+          "readOnly": true,
+          "description": "Gets the disk controller type of the VM captured at the time of the restore point creation."
         }
       },
       "description": "Describes the storage profile."
     },
+    "RestorePointSourceVMDiskControllerType": {
```
Comment (TimLovellSmith) [inline specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/restorePoint.json:729]: Consider that 'Gets the disk controller type of the VM captured at the time of the restore point creation' or 'RestorePointSourceVMDiskControllerType' may be overly specializing your type. Maybe you should just call it 'DiskControllerType'

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/restorePoint.json [hunk 4]
```diff
@@ -717,10 +717,27 @@
...
+      "type": "string",
+      "enum": [
+        "SCSI",
+        "NVMe"
+      ],
+      "x-ms-enum": {
+        "name": "DiskControllerTypes",
```
Comment (TimLovellSmith) [inline specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/restorePoint.json:736]: You should use a singular, unless its a flags enum.

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/restorePoint.json (hunk 3 above)
Comment (TimLovellSmith) [reply specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/restorePoint.json:729]: Looks like this got fixed already

**Pull request merged**